suite/src/benchmark.rs

```diff
@@ -0,0 +1,75 @@
+use core::arch::asm;
+
+/// Returns the machine cycle counter
+#[inline]
+pub fn get_cycle() -> u64 {
```

Comment:
Did you come up with this on your own? If not, can you cite the source? (a link, or a page number in the manual etc)

Reply:
I will also add it to the doc comment.

I used the code provided in Fig. 10.1 on Page 61 of ["Volume I: RISC-V Unprivileged ISA V20191213"](https://riscv.org/wp-content/uploads/2019/12/riscv-spec-20191213.pdf), but I had to modify it to use the control-status-register-read instruction to read the mcycle(h) because we operate in machine mode.

Reply:
Nice, make sure to include the link and your modification and reason as well.